### PR TITLE
Curation activity changes

### PR DESCRIPTION
**Pull request by briri**
Addresses CDL-Dryad/dryad-product-roadmap#176, CDL-Dryad/dryad-product-roadmap#180 and CDL-Dryad/dryad-product-roadmap#185

- Moved association of CurationActivity from Identifier to Resource
- Added methods to Resource to initialize CurationActivity on new record creation, a `curatable?` check and a `curation_status` method which returns the latest CurationActivitiy
- Added a callback to ResourceState that creates a CurationActivity when the resource_state is in_progress or submitted
- Added rake tasks to drop all curation_activities and seed them from existing resource records
- Updated the UI logic to work with the new relationship and hide the status change pencil icon if the the resource is not curatable
- Updated the dataset table's helper to work with status logic
- Updated the status change popup to properly filter out which statuses are available to the user
- Updated controllers to work with new CurationActivity
- Added tests for the new methods on Resource

Some of the filter/search isn't working properly so I need to do some work there. I'd also like to go back and add some more tests to the changes made on Resource and CurationActivity

### CODE REVIEW

File: stash_engine/app/models/stash_engine/concerns/string_enum.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  module Concerns
+
+    module StringEnum
+
+      extend ActiveSupport::Concern
+
+      class_methods do
+
+        # Disabling rubocop here because we reasonably abstract this into callable methods
+        # due to the
```
Comment (ryscher) [inline stash_engine/app/models/stash_engine/concerns/string_enum.rb]: due to the ???

**Comment by ryscher** [comment]
When I try to view the Admin page or create a new dataset, I receive an error. 

There is a note in Slack that this requires running `bundle exec rake identifiers:seed_curation_activities`, but that command also produced an error for me.

File: stash_engine/app/models/stash_engine/concerns/string_enum.rb (hunk 1 above)
Comment (briri) [reply stash_engine/app/models/stash_engine/concerns/string_enum.rb]: Ha ha. Guess I must have gone off on my lunch break there ;)

I broke the one large function into smaller chunks to appease the Rubocop gods. Will remove that comment

**Comment by briri** [comment]
What error did it report @ryscher? Did you run `rake db:migrate` first? It might be failing because the `resource_id` isn't on your `stash_engine_curation_activities` table yet.

**Comment by briri** [comment]
Yes. There is a rake task `identifiers:seed_curation_activities` that deletes all the bad data introduced over the past few months and reseeds the curation_activities table. I've just run this against the dev database (I was working against my local dev since the data model changes would have broken the code currently deployed on dev).

Interested to see which queries you're referring to. I've read and been given recommendations to use 'IN' queries in the past. There does appear to be one in the view that runs once per record (10 times) that could be refactored out to load once in the controller. I'll look into doing that today.

**Pull request merged**